### PR TITLE
IMPROVE - Update the design of MaitenanceEvent list element in DashboardView #174

### LINKED ISSUE #174: IMPROVE - Update the design of `MaitenanceEvent` list element in `DashboardView`
## Describe the feature you want to improve
* Currently the list element for `MaintenanceEvent` the `DashboardView` is getting a bit crowded,
so please update the design like below.
* The placement are as follow:
`Event Tittle` 
`Vehicle Name` + on + `Event Date`
`Event Notes`

image

## Notes
* Please ignore the font/size/weight and just focus on the placement of the text.
* Please do not hard code the date but format it according to locale, languages etc using the `.formatted` API
* You can refer to [Apple document](https://developer.apple.com/documentation/xcode/preparing-dates-numbers-with-formatters/) or [this blog post](https://sarunw.com/posts/new-formatters-in-ios15/) that goes into detail.

## Describe alternatives you've considered
* We discussed making this collapsible but decided on the design☝️

### PR DESCRIPTION
**Pull request by mitaligondaliya**
# What it Does
* Closes https://github.com/mikaelacaron/Basic-Car-Maintenance/issues/174
* Changed the date style as .short

# How I Tested

For example:
* Run the application
* Dashboard view Tap the (+) button
* See the dashboard event list

# Screenshot
Simulator Screenshot - iPhone 15 Plus - 2023-10-31 at 00 29 04

### CODE REVIEW

**Comment by mitaligondaliya** [comment]
@mikaelacaron  You were saying to add the event formatter inside the body? I tried but inside the body we can't assign the dates style So, I kept it as private computed property there.

**Comment by mikaelacaron** [comment]
> @mikaelacaron You were saying to add the event formatter inside the body? I tried but inside the body we can't assign the dates style So, I kept it as private computed property there.

No I mean like this way, from [this article](https://www.hackingwithswift.com/quick-start/swiftui/how-to-format-text-inside-text-views), sorry my other comment looks like I never linked it properly

You put the `formatter` parameter inside the string interpolation, as opposed to using the `string` function that's on the formatter like you did here:

`eventDateFormatter.string(from: event.date)`

File: Basic-Car-Maintenance.xcodeproj/xcshareddata/xcschemes/Basic-Car-Maintenance-WidgetExtension.xcscheme [hunk 1]
```diff
@@ -90,7 +90,6 @@
       savedToolIdentifier = ""
       useCustomWorkingDirectory = "NO"
       debugDocumentVersioning = "YES"
-      askForAppToLaunch = "Yes"
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/xcshareddata/xcschemes/Basic-Car-Maintenance-WidgetExtension.xcscheme]: This shouldn't be removed

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 2]
```diff
@@ -20,21 +20,28 @@ struct DashboardView: View {
...
+    }
+    
     var body: some View {
         NavigationStack {
             List {
                 ForEach(viewModel.searchedEvents) { event in
                     VStack(alignment: .leading, spacing: 8) {
                         Text(event.title)
                             .font(.title3)
+                            .foregroundStyle(.primary)
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: what color is `.primary`? What does changing this do, vs the default?

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
Note the comment I already left about the date formatter, also, the swipe action on the `DashboardView`, the word `Edit` disappeared?

If you have any questions, you can comment directly on that thread, otherwise click the resolve button once you've addressed my comment. Once you fix everything, please click the re-review button

Re-review button

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift (hunk 2 above)
Comment (mitaligondaliya) [reply Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift]: That is default one but not needed I'll remove it

**Comment by mitaligondaliya** [comment]
Screenshot 2023-10-31 at 10 37 46 AM

Swipe action Edit word is there not disappeared.

**Comment by mitaligondaliya** [comment]
@mikaelacaron I've pushed the changes please review it and let me know if there is anything need to change.

File: Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift [hunk 3]
```diff
@@ -20,6 +20,12 @@ struct DashboardView: View {
         viewModel = DashboardViewModel(authenticationViewModel: authenticationViewModel)
     }
     
+    private var eventDateFormat: DateFormatter = {
+            let formatter = DateFormatter()
+            formatter.dateStyle = .short
+            return formatter
+        }()
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Dashboard/Views/DashboardView.swift:27]: The indentation is off, the body of the computed property is indented too far, and the ending brace `}` should be even with the word `private`
I'll fix this manually, don't worry about it

**Review by mikaelacaron — APPROVED** [review]
Great work!! Thanks for making those changes, haha I found the issue for the edit button disappearing for me on a swipe action (lol I'll make this a separate issue)

**Comment by mitaligondaliya** [comment]
@mikaelacaron This one is showing not accepted in hacktoberfest, Could you please mark it as `hacktoberfest-accepted`?

**Comment by mikaelacaron** [comment]
It was merged so it should show as accepted, sometimes it takes a minute, but I did mark it as that too

**Comment by mitaligondaliya** [comment]
> It was merged so it should show as accepted, sometimes it takes a minute, but I did mark it as that too

Okay Thanks 👍🏼 @mikaelacaron

**Pull request merged**